pkg/agent/conversation.go

```diff
@@ -456,13 +478,20 @@ func candidateToShimCandidate(iterator gollm.ChatResponseIterator) (gollm.ChatRe
 					klog.Infof("text is %q", text)
 				} else {
 					yield(nil, fmt.Errorf("no text part found in candidate"))
-					return
+					done = true
+					return false
 				}
 			}
 
 			if _, found := extractJSON(buffer); found {
-				break
+				done = true
+				return false
 			}
+			return true
+		})
+
+		if done {
+			return
```

Comment:
Would be great if we can limit the scope to just the skip-ssl-verify and tackle these as separate PR.

Reply:
@droot I will revert it.

Reply:
Done [force-pushed](https://github.com/GoogleCloudPlatform/kubectl-ai/compare/050f54eaa8efc40b5dd66785f7865fb66b883a5f..d9395b1f84e6b1f61991bfc15b0bda17ab6060f8) the skip branch from [050f54e](https://github.com/GoogleCloudPlatform/kubectl-ai/commit/050f54eaa8efc40b5dd66785f7865fb66b883a5f) to [d9395b1](https://github.com/GoogleCloudPlatform/kubectl-ai/commit/d9395b1f84e6b1f61991bfc15b0bda17ab6060f8)